docs/kb/function-tutorial-deploy.md

```diff
@@ -0,0 +1,212 @@
+```
+or for Windows:
+```
+Env:KUBECONFIG="$HOME\veba\config"    #Windows
+```
+
+
+Kubernetes namespaces are resource boundaries within the cluster.  Function related resources in the VEBA appliance are segregated into the "vmware-functions" namespace.  Use kubectl to list out the resources in the vmware-functions namespace:
+
+```
+kubectl -n vmware-functions get secrets,all
+```
+<img src="./img/kn-ps-slack-vmware-functions.png" width="80%" align="center" class="border m-1 p-1"/>
+
+**A note about kubectl:**  kubectl get "all" does not return "all" resources - only a partial list.  In the above command, secrets are not returned by "all" and so the "secrets" qualifier needs to be added to the command.  The "-n" flag allows specification of the namespace to target.  Without a "-n" flag, the "default" namepace is targeted.
+
+As you can see from the above image, there are some default deployments (default-broker-ingress, sockeye...) but no custom functions yet.  Sockeye displays incoming events and is helpful in troubleshooting.  Sockeye can be accessed by opening a browser to: https://veba-fqdn/events
+
+<img src="./img/kn-ps-slack-events.png" width="80%" align="center" class="border m-1 p-1"/>
+
+
+## Deploy kn-ps-slack Function to the VEBA Appliance
+1.  Move to the /vcenter-event-broker-appliance/examples/knative/powershell/kn-ps-slack directory that you cloned earlier with git.  
+
+2.  Update the slack_secret.json file with your Slack webhook URL.
+
+3.  Create the Kubernetes secret which can then be accessed from within the function by using the environment variable named SLACK_SECRET.
+
+```
+# create secret
+kubectl -n vmware-functions create secret generic slack-secret --from-file=SLACK_SECRET=slack_secret.json
+
+# update label for secret to show up in VEBA UI
+kubectl -n vmware-functions label secret slack-secret app=veba-ui
+```
+
+4.  Edit the function.yaml file with the name of the custom container image you pushed to DockerHub. For the example listed above, we would use:
+
+```
+spec:
+  containers:
+    - image: docker.io/atauber/kn-ps-slack:1.1
+```
+
+By default, the function deployment will filter on the VmPoweredOffEvent vCenter server event. If you wish to change this event type, update the subject field within function.yaml to the desired event type.  The function.yaml file is shown below.  vCenter server events are described here: [vCenter Events](https://vmweventbroker.io/kb/vcenter-events).
+
+<img src="./img/kn-ps-slack-function.png" width="80%" align="center" class="border m-1 p-1"/>
+
+
+5.  Deploy the function to the VEBA Appliance.
+
+```
+# deploy function
+kubectl -n vmware-functions apply -f function.yaml
+```
+
+Now with the new function deployed, we should see these resources in the vmware-functions namespace:
+
+<img src="./img/kn-ps-slack-function3.png" width="80%" align="center" class="border m-1 p-1"/>
+
+
+If we then create a VM named "prod-testvm" and power on / power off, we should have success!
```

Comment:
nit: since the example uses poweredOff let's remove this:

```suggestion
If we then create a VM named "prod-testvm" and power it off, we should have success!
```

Reply:
Fixed.